llvm/lib/Target/AIE/AIEBaseInstrFormats.td

```diff
@@ -34,3 +34,37 @@ class AIEPseudoInstrAttributes {
   let isPseudo = true;
   let isCodeGenOnly = true;
 }
+
+// A helper class to split a dag into two dags. The elements in [0, idx) will
+// be added to DagBeforeSplit, those in [idx + 1, size(DAG)) will be in
+// DagAfterSplit. Note that the element at index idx won't be in either of those.
+class SplitDAGOps<dag DAG, int idx> {
+  list<string> DagKeysBefore = !foreach(i, !range(0, idx), !getdagname(DAG, i));
+  list<DAGOperand> DagArgsBefore = !foreach(key, DagKeysBefore,
+      !getdagarg<DAGOperand>(DAG, key));
+  list<string> DagKeysAfter = !foreach(i, !range(!add(idx, 1), !size(DAG)), !getdagname(DAG, i));
+  list<DAGOperand> DagArgsAfter = !foreach(key, DagKeysAfter,
+      !getdagarg<DAGOperand>(DAG, key));
+  dag DagBeforeSplit = !dag(!getdagop(DAG), DagArgsBefore, DagKeysBefore);
+  dag DagAfterSplit = !dag(!getdagop(DAG), DagArgsAfter, DagKeysAfter);
+}
+
+// A pseudo instruction that has similar properties to RealInst, but the operand
+// at index idx is split into NewSplitIns.
+// Note that the to-be-split operand must be an input.
+class SplitPseudo<Instruction RealInst, int idx, dag NewSplitIns> :
+    AIEBaseInst<(outs), (ins), [], "", "", "">, AIE_HasTiedSubregister, AIEPseudoInstrAttributes {
+
+  let OutOperandList = RealInst.OutOperandList;
+  SplitDAGOps SplitOps = SplitDAGOps<RealInst.InOperandList, !sub(idx, !size(OutOperandList))>;
+  let InOperandList = !con(SplitOps.DagBeforeSplit, NewSplitIns, SplitOps.DagAfterSplit);
+
+  // Copy properties over from RealInst
+  let Constraints = RealInst.Constraints;
+  let mayStore = RealInst.mayStore;
+  let mayLoad = RealInst.mayLoad;
+  let hasSideEffects = RealInst.hasSideEffects;
+  let Defs = RealInst.Defs;
+  let Uses = RealInst.Uses;
+  let Namespace = RealInst.Namespace;
+}
```

Comment:
Perhaps I misunderstood. If there ever is a different access cycle for one of the subregisters, we may want to be able to describe that at least for post-RA sched. It also means that the non-split form will have an inaccuracy in its operand latencies. I worry too much ;-)

Reply:
Currently, the `_split` instructions have no itinerary, this is for two reasons:
 - It would be a lot of work to write or craft so many new itineraries
 - We really do not need scheduling info, as those instructions are seen by neither the premisched, nor the postmisched